pkg/service/jsonencoder/encoder.go

```diff
@@ -0,0 +1,35 @@
+// Copyright (c) 2024 ESET
+// See LICENSE file for redistribution.
+
+package jsonencoder
+
+import (
+	"google.golang.org/protobuf/encoding/protojson"
+	"google.golang.org/protobuf/proto"
+
+	jErrors "github.com/juju/errors"
+)
+
+type Config struct {
+	EmitUnpopulated   bool `mapstructure:"emitUnpopulated"`
+	EmitDefaultValues bool `mapstructure:"emitDefaultValues"`
+}
+
+type Encoder struct {
+	opts protojson.MarshalOptions
+}
+
+func NewOptions(cfg *Config) Encoder {
+	return Encoder{
+		opts: protojson.MarshalOptions{EmitUnpopulated: cfg.EmitUnpopulated, EmitDefaultValues: cfg.EmitDefaultValues},
+	}
+}
+
+func (e Encoder) Format(m proto.Message) (string, error) {
```

Comment:
Optional: Change return value to `([]byte, error)`. This way returned value of `[]byte` can be used to write  HTTP response body in `w.Write()` without need to convert `[]byte` to `string` and then back to `[]byte`.

Reply:
Optional: Rename from `Format` to `Encode`. The name of struct is `Encoder`. Encoder is used for encoding. Formatter is used for formatting.

Compare
```go
jsonResponse, err := jsonEncoder.Format(protoResponse)
```
with
```go
jsonResponse, err := jsonEncoder.Encode(protoResponse)
```